t/t2814-hostlist-cmd.t

```diff
@@ -0,0 +1,148 @@
+	flux hostlist --fallback foo1 >fallback.out &&
+	test "$(cat fallback.out)" = "foo1"
+'
+test_expect_success 'flux-hostlist -c works' '
+	flux hostlist --count  &&
+	test $(flux hostlist --count) -eq 2 &&
+	test $(flux hostlist -c "foo[1-10]") -eq 10
+'
+test_expect_success 'flux-hostlist works with "avail"' '
+	flux resource drain 0 &&
+	test $(flux hostlist --count avail) -eq 1 &&
+	flux resource undrain 0
+'
+test_expect_success 'flux-hostlist works with stdin' '
+	printf "foo1 foo2 foo3"   | flux hostlist - >hl-stdin1.out &&
+	printf "foo1\nfoo2\nfoo3" | flux hostlist - >hl-stdin2.out &&
+	test_debug "grep . hl-stdin*.out" &&
+	printf "foo[1-3]\n" >hl-stdin.expected &&
+	test_cmp hl-stdin.expected hl-stdin1.out &&
+	test_cmp hl-stdin.expected hl-stdin2.out
+'
+test_expect_success 'flux-hostlist works with hostlist args' '
+	flux hostlist "foo[1-3]" > hl-args1.out &&
+	flux hostlist --fallback foo1 foo2 foo3 > hl-args2.out &&
+	printf "foo[1-3]\n" >hl-args.expected &&
+	test_cmp hl-args.expected hl-args1.out &&
+	test_cmp hl-args.expected hl-args2.out
+'
+test_expect_success 'flux-hostlist rejects invalid hostlist' '
+	test_must_fail flux hostlist "foo[1-"
+'
+test_expect_success 'flux-hostlist returns empty hostlist for pending job' '
+	id=$(flux submit --urgency=hold hostname) &&
+	flux hostlist $id >hl-pending.out &&
+	test_must_be_empty hl-pending.out
+'
+test_expect_success 'flux-hostlist --quiet works' '
+	flux hostlist --quiet foo[1-10] > quiet.out &&
+	test_must_be_empty quiet.out &&
+	test_must_fail flux hostlist -q $id
+'
+test_expect_success 'flux-hostlist -e, --expand works' '
+	test_debug "flux hostlist -e foo[1-3]" &&
+	test "$(flux hostlist -e foo[1-3])" = "foo1 foo2 foo3" &&
+	flux hostlist --expand --delimiter="\n" "foo[1-3]" >expand.out &&
+	cat <<-EOF >expand.expected &&
+	foo1
+	foo2
+	foo3
+	EOF
+	test_cmp expand.expected expand.out
+'
+test_expect_success 'flux-hostlist -n, --nth works' '
+	test "$(flux hostlist --nth=1 foo[1-10])" = foo2 &&
+	test "$(flux hostlist --nth=-1 foo[1-10])" = foo10
+'
+test_expect_success 'flux-hostlist -n errors with invalid index' '
+	test_must_fail flux hostlist -n 10 foo[1-10]
+'
+test_expect_success 'flux-hostlist -L, --limit works' '
+	test "$(flux hostlist -L 2 foo[1-10])" = "foo[1-2]" &&
+	test "$(flux hostlist -L -2 foo[1-10])" = "foo[9-10]"
+'
+test_expect_success 'flux-hostlist preserves hostlist order by default' '
+	test "$(flux hostlist host1 host0 host5 host4)" = "host[1,0,5,4]"
+'
+test_expect_success 'flux-hostlist preserves repeated hosts by default' '
+	test "$(flux hostlist host1 host1 host1)" = "host[1,1,1]"
+'
+test_expect_success 'flux-hostlist -S, --sort works' '
+	test "$(flux hostlist -S host3 host2 host1 host0)" = "host[0-3]"
+'
+test_expect_success 'flux-hostlist -S preserves duplicate hosts' '
+	test "$(flux hostlist -S host2 host1 host3 host1)" = "host[1,1-3]"
+'
+test_expect_success 'flux-hostlist -u, --union returns union of all hosts' '
```

Comment:
--union / --unique?

Reply:
Eh, I'll just use `-u` in the description since it is obvious `-u, --union, --unique` are aliases of the same option and we're testing the option behavior not argparse behavior.